### PR TITLE
added option to add to DB

### PR DESCRIPTION
**Pull request by Liamslost**

### CODE REVIEW

**Comment by nadrojb** [comment]
Peer reviewed, looks good to me

File: index.php [hunk 1]
```diff
@@ -2,28 +2,71 @@
...
+        $data = [
+            'state'=> $state,
+            'sublocation'=> $sublocation,
+            'attraction' => $attraction
+        ];
+        addToTable($db, $data);
+    }
+}
+$USATableData = DBgetAllDataUSA($db);
+$All_statesData = DBgetAllDataAll_States($db);
```
Comment (mallowmew) [inline index.php]: It is conventional for variables to start with lowercase after the `$` - these variables are showing up as constants in the github php syntax highlighting. Not a functional problem, just stylistic.

Comment (mallowmew) [reply index.php]: I like that you get all the data at the top of the PHP file and then just call to functions ti display it in the HTML body, very organised & easy to follow.

File: src/DBgetAllData.php [hunk 2]
```diff
@@ -1,5 +1,5 @@
 <?php
-function DBgetAllData ($db)
+function DBgetAllDataUSA ($db)
```
Comment (mallowmew) [inline src/DBgetAllData.php]: It is convention for functions to start with lowercase too - `class`es in Object Oriented programming start with capital letters so the different casing is useful to disambiguate these things & will become important when we start using Object-Oriented PHP later in the course.

`DB` in function names inconsistent with your add function, which also uses the DB. I would say having `DB` in the function names is redundant - kind of obvious these thing suse the DB, but also that's an implementation detail - it's named *how* the function works, when good function names are named for *what the function does*

I also don't like the extra spaces between your function names and the opening parenthesis of the parameters - the parentheses touching the name of the function is also a way to better sight-read the code - if it has a touching paren then "you can just see" that it's a function. 

```suggestion
function dbGetAllDataUSA($db)
```

File: src/DBgetAllData.php [hunk 3]
```diff
@@ -8,4 +8,14 @@ function DBgetAllData ($db)
     $query->execute();
     return $query->fetchall();
 }
-//SELECT `location`, `sublocation`, `attraction`,`info`, `photo`, `rating`  FROM `USA`
+
+// gets all field from All_States table
+function DBgetAllDataAll_States ($db)
```
Comment (mallowmew) [inline src/DBgetAllData.php]: Mix of camelCase and snake_case...

why not `getStates` or `getUSStates`?

File: src/addToTable.php [hunk 4]
```diff
@@ -0,0 +1,18 @@
+<?php
+function addToTable ($db, $data)
```
Comment (mallowmew) [inline src/addToTable.php]:
```suggestion
function addToTable($db, $data)
```

File: src/populateStateDropdown.php [hunk 5]
```diff
@@ -0,0 +1,10 @@
+<?php
+
+function populateStateDropdown(array $statesTable): string
+{
+    $options = '';
+    foreach ($statesTable as $states) {
+        $options .= "<option value=\"{$states['id']}\">{$states['State_name']}</option>";
+    }
+    return $options;
```
Comment (mallowmew) [inline src/populateStateDropdown.php]: You can totally unit test this function, but you haven't added any yet.

**Review by mallowmew — CHANGES_REQUESTED** [review]
You added a function you can unit test!!

File: index.php (hunk 1 above)
Comment (Liamslost) [reply index.php]: done

File: src/DBgetAllData.php (hunk 2 above)
Comment (Liamslost) [reply src/DBgetAllData.php]: that makes sense, thanks!

File: src/DBgetAllData.php (hunk 3 above)
Comment (Liamslost) [reply src/DBgetAllData.php]: done

File: src/addToTable.php (hunk 4 above)
Comment (Liamslost) [reply src/addToTable.php]: done

File: src/populateStateDropdown.php (hunk 5 above)
Comment (Liamslost) [reply src/populateStateDropdown.php]: created success and malformed

File: Tests/displayLocationsTest.php [hunk 6]
```diff
@@ -1,10 +1,10 @@
 <?php
-require_once "src/DisplayLocations.php";
+require_once "src/displayLocations.php";
 use PHPUnit\Framework\TestCase;
 
 
 
-class DisplayLocationsTest extends TestCase{
+class displayLocationsTest extends TestCase{
```
Comment (mallowmew) [inline Tests/displayLocationsTest.php]: classes DO start with capital letters!

So do the files they live in.

File: Tests/populateStateDropdownTest.php [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+
+    }
+
+    public function testpopulateStateDropdownMalformedData(): void
+    {
+        $states = [
+            [
+                false => 'Florida',
+                '2' => 'Georgia',
+            ]];
```
Comment (mallowmew) [inline Tests/populateStateDropdownTest.php]:
```suggestion
            ]
        ];
```

File: index.php [hunk 8]
```diff
@@ -1,29 +1,58 @@
...
 ?>
 <html lang="eng">
-<meta name="viewport" content="width=device-width, initial-scale=1.0" />
 <head>
     <link rel="stylesheet" href="style.css">
 </head>
-<title>Collections App</title>
 
-<body>
+<title>Collections App</title>
```
Comment (mallowmew) [inline index.php]: `<title>` tag goes inside the `<head>`

File: index.php [hunk 9]
```diff
@@ -1,29 +1,58 @@
...
 <nav></nav>
```
Comment (mallowmew) [inline index.php]: `<nav>` Must go inside the body.

**Review by mallowmew — CHANGES_REQUESTED** [review]
Good progress,s ome formatting issues still

**Pull request merged**